backend/src/main/java/com/praksa/auction/service/PersonService.java

```
@@ -1,25 +1,71 @@
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
 import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 
 @Service
 public class PersonService {
     private final PersonRepository personRepositoy;
     @Autowired
-    public PersonService(PersonRepository personRepository){
+    AuthenticationManager authenticationManager;
+    @Autowired
+    JwtUtils jwtUtils;
+    @Autowired
+    PasswordEncoder encoder;
+
+    @Autowired
+    public PersonService(PersonRepository personRepository) {
         this.personRepositoy = personRepository;
     }
-    
-    public Boolean existsByEmail(String email){
+
+    public Boolean existsByEmail(String email) {
         return personRepositoy.existsByEmail(email);
     }
 
-    public void createAccount(Person person) {
-        personRepositoy.save(person);
+    public JwtResponseDto createAccount(RegistrationDto signUpRequest) {
+        if (personRepositoy.existsByEmail(signUpRequest.getEmail())) {
+            String error = "{\"email\":\"This email address is already taken. Please try another one.\"}";
+            throw new IllegalArgumentException(error);
+        }
+        Person p = new Person();
+        p.setEmail(signUpRequest.getEmail());
+        p.setFirstName(signUpRequest.getFirstName());
+        p.setLastName(signUpRequest.getLastName());
+        p.setPassword(encoder.encode(signUpRequest.getPassword()));
+        personRepositoy.save(p);
+        return logIn(new LogInDto(signUpRequest.getEmail(), signUpRequest.getPassword()));
+    }
+
+    public Person getByEmail(String email) {
+        return personRepositoy.findByEmail(email).get();
+    }
+
+    public JwtResponseDto logIn(LogInDto loginInfo) {
+        if (!personRepositoy.existsByEmail(loginInfo.getEmail())) {
+            String error = "{\"email\":\"Email address not found\"}";
```

No need to make json, do `throw new UsernameNotFoundException("Email address not found");`, and in controller in catch you can return `LogInFailResponse` that will contain error message. E.g. 
`return new ResponseEntity<>(response, HttpStatus.BAD_REQUEST);`